data_engineering/cmpny_bc_cls/utils/feature_imputation.py

```diff
@@ -2,5 +2,18 @@
 
 
 def fill_null_with_zeros(df: pd.DataFrame, columns: list) -> pd.DataFrame:
-    df[columns] = df[columns].fillna(0)
-    return df
+    df_new = df.copy()
+    df_new[columns] = df_new[columns].fillna(0)
+    return df_new
+
+
+def fill_null_with_mean(df: pd.DataFrame, columns: list) -> pd.DataFrame:
+    df_new = df.copy()
+    df_new[columns] = df_new[columns].fillna(df[columns].mean())
+    return df_new
+
+
+def fill_null_with_midean(df: pd.DataFrame, columns: list) -> pd.DataFrame:
```

Comment:
btw can you change midean to median please 🤓

Reply:
hahahah yeah nice spot